### DISCUSSION TITLE
Official support for the new "flat config" ESLint approach

### DISCUSSION THREAD

**Discussion by hb20007** [vercel/next.js#63999] [Ideas]
### Goals

I would like to see the [new ESLint "flat config"](https://eslint.org/docs/latest/use/configure/configuration-files-new) officially supported for the `next/core-web-vitals` ESLint config.  There is already an [answered discussion](https://github.com/vercel/next.js/discussions/49337) regarding this, but the final comment left by the OP is the following:

> Switching my projects to flat config was taking more time than it was worth.

So, the issue remains unsolved. I would like to see official support on how `"extends": [""next/core-web-vitals"]` in `.eslintrc.json` can simply be migrated to a single line in `eslint.config.js`. 

### Background

ESLint is [moving](https://eslint.org/docs/latest/use/configure/migration-guide) from `.eslintrc.json` to `eslint.config.js`, and many other shareable configs (e.g., `typescript-eslint`) already support this.

### Proposal

[This documentation page](https://nextjs.org/docs/app/building-your-application/configuring/eslint) will also need to be updated, showing the option of using `eslint.config.js`.

**Comment by hb20007** [comment]
Issue: https://github.com/vercel/next.js/issues/64114

**Comment by ryancwalsh** [comment]
I'd really love to see Next.js support the flat config. +1
